File: summer/WebContent/js/system/windows/data/UpdateSourceTrigger.js
/**
 * Second Check 12-27
 * UpdateSourceTrigger
 */
define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	
    var UpdateSourceTrigger = declare("UpdateSourceTrigger", Object,{ 
    });	
    
    /**
     * Obtain trigger from target property default 
     */
    UpdateSourceTrigger.Default = 0;

	/**
	 * Update whenever the target property changes
	 */
    UpdateSourceTrigger.PropertyChanged = 1;
    
    /**
     * Update only when target element loses focus, or when Binding deactivates
     */ 
    UpdateSourceTrigger.LostFocus = 2; 

	/**
	 * Update only by explicit call to BindingExpression.UpdateSource()
	 */
    UpdateSourceTrigger.Explicit = 3;
    
    UpdateSourceTrigger.Type = new Type("UpdateSourceTrigger", UpdateSourceTrigger, [Object.Type]);
    return UpdateSourceTrigger;
});

File: org.summer.view.widget/src/org/summer/view/widget/data/UpdateSourceTrigger.java
package org.summer.view.widget.data;
public enum UpdateSourceTrigger 
    {
        /// <summary> Obtain trigger from target property default </summary>
        Default,
        /// <summary> Update whenever the target property changes </summary> 
        PropertyChanged,
        /// <summary> Update only when target element loses focus, or when Binding deactivates </summary> 
        LostFocus, 
        /// <summary> Update only by explicit call to BindingExpression.UpdateSource() </summary>
        Explicit 
    }

    /// <summary>
    /// Base class for Binding, PriorityBinding, and MultiBinding. 
    /// </summary>
//    [MarkupExtensionReturnType(typeof(object))] 
//    [Localizability(LocalizationCategory.None, Modifiability = Modifiability.Unmodifiable, Readability = Readability.Unreadable)] // Not localizable by-default 
    